OPHD/Things/Robots/Robot.cpp

```diff
@@ -1,12 +1,14 @@
 #include "Robot.h"
 
-Robot::Robot(const std::string& name, const std::string& sprite_path) :
-	Thing(name, sprite_path, "running")
+Robot::Robot(const std::string& name, const std::string& sprite_path, Type t) :
+	Thing(name, sprite_path, "running"),
+	mType{ t }
```

Comment:
We should really be avoiding one letter variable names like `t`, particularly for names/types that are not universal (like `x`, `y`, or `i`). It's not like source code storage size is an issue in this day and age. And thinking speed is far more limiting than typing speed when it comes to programming.

Reply:
't' kind of is but generally speaking I agree. Using 'type' works but it's also a keyword and I would rather avoid '_type'.

Reply:
Perhaps I'm confusing it with `typedef` and `typeid`. Did a check for reserved keywords and `type` isn't part of that. I guess I have no excuse to fully expand it now. :D

Reply:
I was actually also questioning if `type` was a reserved keyword when I first looked at the code. That's probably why I was a bit vague and didn't suggest renaming it to `type`. :wink:

Only noticed just as I was closing the tab that there was a method named `type()`.